js/components/graph/EdgeGroup.js

```diff
@@ -11,31 +13,52 @@ export default class EdgeGroup extends React.Component {
     this.state = {};
   }
 
-  // When an edge's state changes and the edge is not undefined,
-  // it will call updateEdgeStatus and update EdgeGroup's state with its
-  // edgeID and status. This function is passed as a props to Edge.
+  /**
+   * Updates EdgeGroup's state with the state of an Edge when the Edge's state changes.
+   * This function is passed as a property to Edge.
+   * @param {string} edgeID
+   * @param {string} state
+   */
   updateEdgeStatus = (edgeID, state) => {
     var isMissing = state === "missing";
     this.setState({ [edgeID]: isMissing });
   }
 
+  /**
+   * After each render after the initial, update the status of each Edge
+   */
   componentDidUpdate() {
     this.props.edgesJSON.forEach(edgeJSON => {
       this[edgeJSON.id_].updateStatus();
     });
   }
 
+  /**
+   * Set all Edges to the inactive status
+   */
   reset = () => {
     this.props.edgesJSON.forEach(edgeJSON => {
       this[edgeJSON.id_].setState({ status: "inactive" });
     });
   }
-
+  /**
+   * This function is used as a callback ref. See {@link:https://reactjs.org/docs/refs-and-the-dom.html#callback-refs}
+   * @param {JSON} edgeJSON Represents a single edge
+   * @returns {function} A function that adds/updates a key value pair to EdgeGroup of an Edge's ID
+   *  and the Edge object
+   */
   setRefEntry = edgeJSON =>{
```

Comment:
I think this function is very confusing, especially for beginners to Javascript. I think it could be easily reimplemented as an if-else statement following the idea below : 
* if `elem` is null, add the key value pair
* otherwise, don't do anything

In my opinion, this would be easier to read. Unless this is a standard pattern for JS that is good for SDS members to be exposed to, I think this should be refactored.

Reply:
This is a common programming idiom (not just in Javascript). It's good that you encountered it, but it doesn't warrant a special comment. Please remove the inline comments you added here.

Reply:
Oh ok, sounds good. Will remove. Thank you!